engine/src/main/java/org/pentaho/di/www/PrepareExecutionTransServlet.java

```diff
@@ -39,13 +39,15 @@
 import org.pentaho.di.trans.Trans;
 import org.pentaho.di.trans.TransConfiguration;
 import org.pentaho.di.trans.TransExecutionConfiguration;
+import org.pentaho.di.www.cache.TransJobStatusCache;
 
 
 public class PrepareExecutionTransServlet extends BaseHttpServlet implements CartePluginInterface {
   private static Class<?> PKG = PrepareExecutionTransServlet.class; // for i18n purposes, needed by Translator2!!
 
   private static final long serialVersionUID = 3634806745372015720L;
   public static final String CONTEXT_PATH = "/kettle/prepareExec";
+  private TransJobStatusCache cache = TransJobStatusCache.getInstance();
```

Comment:
Since servlets are singletons, I don't think you should do this.

Reply:
normally, instance variables in a servlet are no-nos. In this case it's likely OK.